bark/world/evaluation/safe_distances/evaluator_safe_dist_drivable_area.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2020 fortiss GmbH
+//
+// Authors: Julian Bernhard, Klemens Esterle, Patrick Hart and
+// Tobias Kessler
+//
+// This work is licensed under the terms of the MIT license.
+// For a copy, see <https://opensource.org/licenses/MIT>.
+
+#include "bark/world/observed_world.hpp"
+#include "bark/world/evaluation/safe_distances/evaluator_safe_dist_drivable_area.hpp"
+
+namespace bark {
+namespace world {
+namespace evaluation {
+
+using bark::geometry::Point2d;
+using bark::geometry::Polygon;
+using bark::geometry::Pose;
+using bark::models::dynamic::State;
+using bark::models::dynamic::StateDefinition;
+
+EvaluatorSafeDistDrivableArea::EvaluatorSafeDistDrivableArea(const bark::commons::ParamsPtr& params, const AgentId& agent_id) :
```

Comment:
so the purpose of this evaluator is to check if there is a safe longitudinal distance towards the end of the driving corridor? but what do you check for laterally? is it a safe distance or is it only a collision check to a buffered distance?

Reply:
You can specify a separate offset or vehicle front and side, this gives an inflated ego agent shape which is checked for collision with the road corridor